### DISCUSSION TITLE
Suggestion:  Grid columns built in (tiny change for huge functionality)

### DISCUSSION THREAD

**Discussion by ethanpil** [picocss/pico#582] [Ideas]
I know officially Pico documentation reads 

> As Pico focuses on native HTML elements, we kept this grid system minimalist.
> 
> A complete grid system in flexbox, with all the ordering, offsetting, and breakpoints utilities, can be heavier than the total size of the Pico library. Not really in the Pico spirit.

I would like to propose the following tiny addition to Pico which provides probably 99% of the use cases for grid type layouts without _"all the ordering, offsetting, and breakpoints utilities"_ were trying to avoid.

The issues I had with the current implementation is that each row needs a new div and its very cumbersome when generating repeating blocks. Additionally many layouts need some variation of grid columns...

Here is a [demo link](https://jsfiddle.net/ethanpil/qaLwc9jh/).

I would love to know your thoughts. Since I dont want to recompile Pico I use them as a second class like this: `<div class="grid grid-4">...</div>` but certainly we can eliminate the need for the first `grid` class if that makes more sense.

Can make a PR if the community is interested. @lucaslarroche :)

```
.grid-2 {
   grid-template-columns: repeat(2, 1fr);
}
.grid-3 {
   grid-template-columns: repeat(3, 1fr);
}
.grid-4 {
   grid-template-columns: repeat(4, 1fr);
}
.grid-5 {
   grid-template-columns: repeat(5, 1fr);
}
.grid-6 {
   grid-template-columns: repeat(6, 1fr);
}
.grid-8 {
   grid-template-columns: repeat(8, 1fr);
}

@media only screen and (max-width: 768px) {
  .grid-2, .grid-3, .grid-4, .grid-5, .grid-6, .grid-8 {
     grid-template-columns: repeat(1, 1fr);
  }
}
```

**Comment by Deufel** [comment]
Just a follow up to this another usefull simple grid change is.... 
```.css
grid-template-columns: repeat(auto-fill, minmax(240px, 1fr))
```
Note its helpful to remove the bottom margin from the items in the grid

**Comment by ehausen** [comment]
I think it perhaps would be more in the spirit of Pico to define your columns per site or elements basis without helper class.

**Reply by claudiob** [reply]
I use grids a lot with Pico but my needs are different than a symmetric 50/50 split (2 columns) or 25/25/25/25 split (4 columns)…

Since there are many variations of the grid (like the one @Deufel pointed out) I think @ehausen is making the right call on this one.

**Comment by Seb-C** [comment]
Another minimalist approach is to use variables and the style attribute.
It results in a similar syntax, but much more smaller and flexible css:
```css
.grid {
	--col-l: 3;
	--col-m: 2;
	--col-s: 1;

	grid-template-columns: repeat(var(--col-l), 1fr);
	@media only screen and (max-width: 992px) {
		grid-template-columns: repeat(var(--col-m), 1fr);
	}
	@media only screen and (max-width: 768px) {
		grid-template-columns: repeat(var(--col-s), 1fr);
	}
}
```

```html
<div class="grid" style="--col-s: 1; --col-m: 3; --col-l: 6">
	<div>A</div>
	<div>B</div>
	<div>C</div>
	<div>D</div>
	<div>E</div>
	<div>F</div>
</div>
```
